src/app/model/model.service.ts

```diff
@@ -0,0 +1,67 @@
+import { Injectable } from '@angular/core';
+import { Storage } from '@ionic/storage';
+
+import { Serialize, Deserialize } from 'cerialize';
+
+import { Model } from './model';
+
+/**
+ * A generic ModelService which stores models in a Ionic Storage.
+ *
+ * Ionic Storage is just a simple key/value storage, which means this will store
+ * all the model information at the same place. This is obviously not ideal.
+ * On the upside Ionic Storage is highly portable and should work on all devices
+ * as well as in a web browser without cordova support.
+ *
+ * ModelService implements basic separation of models using constructor names.
+ */
+export class ModelService<M extends Model> {
```

Comment:
fragemi öbs nid besser wär wes nume ei `ModelService` gäb u die angere services bruche dä statt drvo abzleite.

Reply:
auso composition vs inheritance? I finge hier isch ganz klar d `is-a` relation gae drum hani mi fuer ableite entschide.

Reply:
Refactored the code big times. We now have two different kind of services.
- Normal services in the `/services/` directory. Those services provide business functionality
- Repository services in the `/model(s)` directory. These services provide access to the persistence-layer and additional data transformation for model data.